src/components/top-bar-equipment-seach-dialog/top-bar-equipment-search-popover.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { EquipmentType } from '@gridsuite/commons-ui';
+import { HorizontalRule } from '@mui/icons-material';
+import {
+    Box,
+    FormControl,
+    FormControlLabel,
+    Popover,
+    Radio,
+    RadioGroup,
+} from '@mui/material';
+import { Dispatch, SetStateAction } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+const unsearchableTypes = [
+    EquipmentType.SWITCH,
+    EquipmentType.VSC_CONVERTER_STATION,
+    EquipmentType.LCC_CONVERTER_STATION,
+];
+
+const styles = {
+    popoverPaper: {
+        // prevent prevent component from hovering searchbar when too high
+        maxHeight: '45%',
+    },
+    noFilterOption: {
+        display: 'flex',
+        gap: 1,
+        paddingBottom: 1,
+        cursor: 'pointer',
+    },
+};
+
+interface TopBarEquipmentSearchPopoverProps {
+    open: boolean;
+    setIsOpen: Dispatch<SetStateAction<boolean>>;
+    anchorEl: HTMLDivElement | null;
+    equipmentType: EquipmentType | null;
+    setEquipmentType: Dispatch<SetStateAction<EquipmentType | null>>;
+}
+
+export const TopBarEquipmentSearchPopover = (
+    props: TopBarEquipmentSearchPopoverProps
+) => {
+    const { open, setIsOpen, equipmentType, setEquipmentType, anchorEl } =
+        props;
+
+    return (
+        <Popover
+            slotProps={{
+                paper: {
+                    style: styles.popoverPaper,
+                },
+            }}
+            anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'right',
+            }}
+            transformOrigin={{
+                vertical: 'top',
+                // can't use theme.spacing in a clean way here because it would return a string which is not accepted by "horizontal"
+                horizontal: 32,
```

Comment:
```suggestion
                horizontal: parseInt(theme.spacing(2), 10),
```

Reply:
it does not work
the result of "theme.spacing" is something like "16 px"
to make it work you have to use substr to remove "px" in the end
I'm not sure we want to do it ?
@EstherDarkish what do you think ?

Reply:
The less complex, the better. I like the hard value because there is a comment.